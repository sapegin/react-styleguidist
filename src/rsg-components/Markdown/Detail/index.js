export { default } from 'rsg-components/Markdown/Detail/DetailRenderer';
export { default as DetailSummary } from 'rsg-components/Markdown/Detail/DetailSummaryRenderer';
