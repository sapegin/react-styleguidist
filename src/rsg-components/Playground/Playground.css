.root {
	composes: border from "../../styles/colors.css";
	margin-bottom: 30px;
	border-radius: 3px;
}

.editor {
}

.preview {
	margin-bottom: 3px;
	padding: 15px;
}

.codeToggle {
	composes: font from "../../styles/common.css";
	composes: border from "../../styles/colors.css";
	composes: link from "../../styles/colors.css";
	cursor: pointer;
	display: inline-block;
	float: right;
	margin-bottom: -30px;
	margin-right: -1px;
	border-bottom-left-radius: 4px;
	border-bottom-right-radius: 4px;
	border-top: 0;
	padding: 4px 8px;
	font-size: small;
}

.showCode {
	composes: codeToggle;
	background: #fafafa;
}

.hideCode {
	composes: codeToggle;
	background: #f5f5f5;
}
