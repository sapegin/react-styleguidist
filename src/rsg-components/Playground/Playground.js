import React, { Component, PropTypes } from 'react';
import Editor from 'rsg-components/Editor';
import Preview from 'rsg-components/Preview';

import s from './Playground.css';

export default class Playground extends Component {
	static propTypes = {
		highlightTheme: PropTypes.string.isRequired,
		code: PropTypes.string.isRequired,
		evalInContext: PropTypes.func.isRequired,
		showCode: PropTypes.bool,
	};

	constructor(props) {
		super();
		this.state = {
			code: props.code,
			showCode: false,
		};
	}

	componentWillReceiveProps(nextProps) {
		let { code } = nextProps;
		if (code) {
			this.setState({
				code,
			});
		}
	}

	handleChange(newCode) {
		this.setState({
			code: newCode,
		});
	}

	handleCodeToggle() {
		this.setState({
			showCode: !this.state.showCode,
		});
	}

	render() {
		let { code } = this.state;
		let { highlightTheme } = this.props;

		return (
			<div className={s.root}>
				<div className={s.preview}>
					<Preview code={code} evalInContext={this.props.evalInContext} />
				</div>
				{this.state.showCode ? (
					<div className={s.editor}>
						<Editor code={code} highlightTheme={highlightTheme} onChange={code => this.handleChange(code)} />
						<div className={s.hideCode} onClick={() => this.handleCodeToggle()}>
							hide code
						</div>
					</div>
				) : (
					<div className={s.showCode} onClick={() => this.handleCodeToggle()}>
						show code
					</div>
				)}
			</div>
		);
	}
}
