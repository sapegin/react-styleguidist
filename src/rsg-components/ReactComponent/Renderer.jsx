import { PropTypes } from 'react';

const s = require('./ReactComponent.css');

const Renderer = ({ name, pathLine, description, propList, examples }) => {
	return (
		<div className={s.root}>
			<header className={s.header}>
				<h2 className={s.heading} id={name}>
					{name}
				</h2>
				<div className={s.pathLine}>{pathLine}</div>
			</header>
			<div>
				{description}
			</div>
			{propList}
			{examples}
		</div>
	);
};

Renderer.propTypes = {
	name: PropTypes.string.isRequired,
	pathLine: PropTypes.string.isRequired,
	description: PropTypes.object,
	propList: PropTypes.object,
	examples: PropTypes.array,
};


export default Renderer;
