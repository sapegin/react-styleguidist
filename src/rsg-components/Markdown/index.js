import { asArrayMarkdown } from './Markdown';

export { default, asArrayMarkdown } from './Markdown.js';

