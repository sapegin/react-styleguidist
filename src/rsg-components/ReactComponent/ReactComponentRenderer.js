import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';
import Markdown from 'rsg-components/Markdown';

const styles = ({ color, fontSize, fontFamily, space }) => ({
	root: {
		marginBottom: space[6],
	},
	header: {
		marginBottom: space[3],
	},
	tabs: {
		marginBottom: space[3],
	},
	tabButtons: {
		marginBottom: space[1],
	},
	pathLine: {
		fontFamily: fontFamily.monospace,
		color: color.light,
		fontSize: fontSize.small,
	},
	docs: {
		color: color.base,
		fontSize: fontSize.text,
	},
});

export function ReactComponentRenderer({
	classes,
	name,
	heading,
	pathLine,
	description,
	docs,
	examples,
	tabButtons,
	tabBody,
}) {
	return (
		<div className={classes.root} id={name + '-container'}>
			<header className={classes.header}>
				{heading}
				<div className={classes.pathLine}><Markdown text={pathLine} /></div>
			</header>
			{(description || docs) &&
				<div className={classes.docs}>
					{description}
					{docs}
				</div>}
			{tabButtons &&
				<div className={classes.tabs}>
					<div className={classes.tabButtons}>{tabButtons}</div>
					{tabBody}
				</div>}
			{examples}
		</div>
	);
}

ReactComponentRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	name: PropTypes.string.isRequired,
	heading: PropTypes.node.isRequired,
	pathLine: PropTypes.string.isRequired,
	tabButtons: PropTypes.node,
	tabBody: PropTypes.node,
	description: PropTypes.node,
	docs: PropTypes.node,
	examples: PropTypes.node,
	isolated: PropTypes.bool,
};

export default Styled(styles)(ReactComponentRenderer);
