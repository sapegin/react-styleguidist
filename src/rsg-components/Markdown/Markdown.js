import React from 'react';
import PropTypes from 'prop-types';
import { compiler } from 'markdown-to-jsx';
import mapValues from 'lodash/mapValues';
// import memoize from 'lodash/memoize';
import Styled from 'rsg-components/Styled';
import Link from 'rsg-components/Link';
import { styles as paraStyles } from 'rsg-components/Para';
import MarkdownHeading from 'rsg-components/Markdown/MarkdownHeading';

// We’re explicitly specifying Webpack loaders here so we could skip specifying them in Webpack configuration.
// That way we could avoid clashes between our loaders and user loaders.
// eslint-disable-next-line import/no-unresolved
require('!!../../../loaders/style-loader!../../../loaders/css-loader!highlight.js/styles/tomorrow.css');

// Temporary disable memoization to fix: https://github.com/styleguidist/react-styleguidist/issues/348
// TODO: Remove after merge: https://github.com/probablyup/markdown-to-jsx/pull/96
const memoize = a => a;

// Code blocks with server-side syntax highlight
function Code({ children, className }) {
	const isHighlighted = className && className.indexOf('lang-') !== -1;
	if (isHighlighted) {
		return <code className={className} dangerouslySetInnerHTML={{ __html: children }} />;
	}
	return <code className={className}>{children}</code>;
}
Code.propTypes = {
	children: PropTypes.node,
	className: PropTypes.string,
};

// Custom CSS classes for each tag: <em> → <em className={s.em}> + custom components
const getBaseOverrides = memoize(classes => {
	const styleOverrides = mapValues(classes, value => ({
		props: {
			className: value,
		},
	}));

	return {
		...styleOverrides,
		a: {
			component: Link,
		},
		h1: {
			component: MarkdownHeading,
			props: {
				level: 1,
			},
		},
		h2: {
			component: MarkdownHeading,
			props: {
				level: 2,
			},
		},
		h3: {
			component: MarkdownHeading,
			props: {
				level: 3,
			},
		},
		h4: {
			component: MarkdownHeading,
			props: {
				level: 4,
			},
		},
		h5: {
			component: MarkdownHeading,
			props: {
				level: 5,
			},
		},
		h6: {
			component: MarkdownHeading,
			props: {
				level: 6,
			},
		},
		code: {
			component: Code,
			props: {
				className: classes.code,
			},
		},
	};
}, () => 'getBaseOverrides');

// Inline mode: replace <p> (usual root component) with <span>
const getInlineOverrides = memoize(classes => {
	const overrides = getBaseOverrides(classes);

	return {
		...overrides,
		p: {
			component: 'span',
			props: {
				className: classes.base,
			},
		},
	};
}, () => 'getInlineOverrides');

const styles = ({ space, fontFamily, fontSize, color, borderRadius }) => ({
	base: {
		color: color.base,
		fontFamily: fontFamily.base,
		fontSize: 'inherit',
	},
	para: paraStyles({ space, color, fontFamily }).para,
	p: {
		composes: '$para',
	},
	ul: {
		composes: '$para',
		paddingLeft: space[3],
	},
	ol: {
		composes: '$para',
		listStyleType: 'decimal',
		paddingLeft: space[3],
	},
	li: {
		composes: '$base',
		listStyleType: 'inherit',
	},
	input: {
		color: color.base,
		display: 'inline-block',
		margin: [[0, '0.35em', '0.25em', '-1.2em']],
		verticalAlign: 'middle',
	},
	blockquote: {
		composes: '$para',
		fontSize: fontSize.base,
		margin: [[space[2], space[4]]],
		padding: 0,
	},
	hr: {
		composes: '$para',
		borderWidth: [[0, 0, 1, 0]],
		borderColor: color.border,
		borderStyle: 'solid',
	},
	em: {
		composes: '$base',
		fontStyle: 'italic',
	},
	strong: {
		composes: '$base',
		fontWeight: 'bold',
	},
	code: {
		fontFamily: fontFamily.monospace,
		fontSize: 'inherit',
		color: 'inherit',
		background: 'transparent',
		whiteSpace: 'inherit',
	},
	pre: {
		composes: '$para',
		backgroundColor: color.codeBackground,
		border: [[1, color.border, 'solid']],
		padding: [[space[1], space[2]]],
		fontSize: fontSize.small,
		borderRadius,
		whiteSpace: 'pre',
	},
	table: {
		composes: '$para',
		borderCollapse: 'collapse',
	},
	thead: {
		composes: '$hr',
	},
	tbody: {},
	td: {
		fontFamily: fontFamily.base,
		padding: [[space[0], space[2], space[0], 0]],
		fontSize: fontSize.base,
	},
	th: {
		composes: '$td',
		fontWeight: 'bold',
	},
	tr: {},
});

function Markdown({ classes, text, inline }) {
	const overrides = inline ? getInlineOverrides(classes) : getBaseOverrides(classes);
	return compiler(text, { overrides });
}

Markdown.propTypes = {
	classes: PropTypes.object.isRequired,
	text: PropTypes.string.isRequired,
	inline: PropTypes.bool,
};

export default Styled(styles)(Markdown);
