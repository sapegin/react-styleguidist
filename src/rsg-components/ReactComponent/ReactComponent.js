import React, { PropTypes } from 'react';
import Markdown from 'rsg-components/Markdown';
import Props from 'rsg-components/Props';
import Methods from 'rsg-components/Methods';
import Examples from 'rsg-components/Examples';
import ReactComponentRenderer from 'rsg-components/ReactComponent/ReactComponentRenderer';

const ExamplePlaceholder = process.env.NODE_ENV === 'development'
	? require('rsg-components/ExamplePlaceholder').default
	: () => <div></div>
;

export default function ReactComponent({ component }, { isolatedComponent = false }) {
	const { name, slug, pathLine } = component;
	const { description, props, examples, methods } = component.props;
	if (!name) {
		return null;
	}

	return (
		<ReactComponentRenderer
			name={name}
			slug={slug}
			pathLine={pathLine}
			description={description && <Markdown text={description} />}
			props={props && <Props props={props} />}
			methods={methods.length > 0 && <Methods methods={methods} />}
			examples={examples.length > 0
				? <Examples examples={examples} name={name} />
				: <ExamplePlaceholder name={name} />
			}
			isolated={isolatedComponent}
		/>
	);
}

ReactComponent.propTypes = {
	component: PropTypes.object.isRequired,
};

ReactComponent.contextTypes = {
	isolatedComponent: PropTypes.bool,
};
