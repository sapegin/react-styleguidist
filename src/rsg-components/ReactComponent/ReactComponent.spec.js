import React from 'react';
import noop from 'lodash/noop';
import Markdown from '../Markdown';
import Playground from '../Playground';
import ReactComponent from './ReactComponent';
import { ReactComponentRenderer } from './ReactComponentRenderer';

const component = {
	name: 'Foo',
	slug: 'foo',
	pathLine: 'foo/bar.js',
	props: {
		description: 'Bar',
		methods: [],
		examples: [],
	},
	metadata: {},
};
const componentWithExamples = {
	name: 'Foo',
	slug: 'foo',
	pathLine: 'foo/bar.js',
	props: {
		description: 'Bar',
		methods: [],
		examples: [
			{
				type: 'code',
				content: '<button>OK</button>',
				evalInContext: noop,
			},
			{
				type: 'markdown',
				content: 'Hello *world*!',
			},
		],
	},
	metadata: {},
};
const componentWithProps = {
	name: 'Foo',
	slug: 'foo',
	pathLine: 'foo/bar.js',
	props: {
		description: 'Bar',
		props: {
			children: {
				type: { name: 'string' },
				required: true,
				description: 'Button label.',
			},
		},
		methods: [],
		examples: [],
	},
	metadata: {},
};
const componentWithMethods = {
	name: 'Foo',
	slug: 'foo',
	pathLine: 'foo/bar.js',
	props: {
		description: 'Bar',
		methods: [
			{
				name: 'set',
				params: [
					{
						name: 'newValue',
						description: 'New value for the counter.',
						type: { name: 'Number' },
					},
				],
				returns: null,
				description: 'Sets the counter to a particular value.',
			},
		],
		examples: [],
	},
	metadata: {},
};

it('should render component renderer with example placeholder', () => {
	const actual = shallow(
		<ReactComponent
			component={component}
		/>
	);

	expect(actual).toMatchSnapshot();
});

it('should render component renderer for component with examples', () => {
	const actual = shallow(
		<ReactComponent
			component={componentWithExamples}
		/>
	);

	expect(actual).toMatchSnapshot();
});

it('should render component renderer for component with props', () => {
	const actual = shallow(
		<ReactComponent
			component={componentWithProps}
		/>
	);

	expect(actual).toMatchSnapshot();
});

it('should render component renderer for component with methods', () => {
	const actual = shallow(
		<ReactComponent
			component={componentWithMethods}
		/>
	);

	expect(actual).toMatchSnapshot();
});

it('should return null when component has no name', () => {
	const actual = shallow(
		<ReactComponent component={{ slug: 'foo', props: {} }} />
	);

	expect(actual.node).toBe(null);
});

it('renderer should render component', () => {
	const actual = shallow(
		<ReactComponentRenderer
			classes={{}}
			name={componentWithExamples.name}
			slug={componentWithExamples.slug}
			pathLine={componentWithExamples.pathLine}
			description={componentWithExamples.props.description}
			examples={[
				<Playground
					key={0}
					index={0}
					name="Component"
					code={componentWithExamples.props.examples[0].content}
					evalInContext={componentWithExamples.props.examples[0].evalInContext}
				/>,
				<Markdown
					key={1}
					text={componentWithExamples.props.examples[1].content}
				/>,
			]}
			metadata={{}}
		/>
	);

	expect(actual).toMatchSnapshot();
});

test('should render component not in the isolation mode by default', () => {
	const actual = shallow(
		<ReactComponentRenderer
			classes={{}}
			name="Test"
			slug="test"
			pathLine="test"
			metadata={{}}
		/>
	);

	expect(actual).toMatchSnapshot();
});

test('should render component in isolation mode', () => {
	const actual = shallow(
		<ReactComponentRenderer
			classes={{}}
			name="Test"
			slug="Test"
			pathLine="test"
			metadata={{}}
			isolated
		/>
	);

	expect(actual).toMatchSnapshot();
});

test('should render props section', () => {
	const actual = shallow(
		<ReactComponentRenderer
			classes={{}}
			name="Test"
			slug="test"
			pathLine="test"
			props={<div>test</div>}
			metadata={{}}
		/>
	);

	expect(actual).toMatchSnapshot();
});

test('should render methods section', () => {
	const actual = shallow(
		<ReactComponentRenderer
			classes={{}}
			name="Test"
			slug="test"
			pathLine="test"
			props={null}
			methods={<div>test</div>}
			metadata={{}}
		/>
	);

	expect(actual).toMatchSnapshot();
});

test('should render both props and methods section', () => {
	const actual = shallow(
		<ReactComponentRenderer
			classes={{}}
			name="Test"
			slug="test"
			pathLine="test"
			props={<div>prop</div>}
			methods={<div>method</div>}
			metadata={{}}
		/>
	);

	expect(actual).toMatchSnapshot();
});

test('should not render props / methods section if there is no content', () => {
	const actual = shallow(
		<ReactComponentRenderer
			classes={{}}
			name="Test"
			slug="test"
			pathLine="test"
			metadata={{}}
		/>
	);

	expect(actual).toMatchSnapshot();
});

it('renderer should render name with strike-through style when @deprecated is present in tags', () => {
	const deprecatedComponent = {
		...component,
		props: {
			...component.props,
			tags: {
				deprecated: [{
					title: 'deprecated',
					description: 'I am deprecated',
				}],
			},
		},
	};

	const actual = shallow(
		<ReactComponent
			component={deprecatedComponent}
		/>
	);

	expect(actual).toMatchSnapshot();
});
