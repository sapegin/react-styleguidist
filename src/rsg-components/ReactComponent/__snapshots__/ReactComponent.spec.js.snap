// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer should render component 1`] = `
<div
  id="Foo-container"
>
  <header>
    <_class
      className=""
      level={2}
      slug="foo"
    >
      Foo
    </_class>
    <div>
      foo/bar.js
    </div>
    <div>
      <_class
        href="#!/Foo"
      >
        Open isolated ⇢
      </_class>
    </div>
  </header>
  <div>
    <_class
      text="Bar"
    />
    <JsDoc />
  </div>
  <Playground
    code="<button>OK</button>"
    evalInContext={[Function]}
    index={0}
    name="Component"
  />
  <_class
    text="Hello *world*!"
  />
</div>
`;

exports[`renderer should render name with strike-through style when @deprecated is present in tags 1`] = `
<_class
  description="Bar"
  examples={
    <Component
      name="Foo"
    />
  }
  isolated={false}
  metadata={Object {}}
  methods={false}
  name="Foo"
  pathLine="foo/bar.js"
  slug="foo"
  tags={
    Object {
      "deprecated": Array [
        Object {
          "description": "I am deprecated",
          "title": "deprecated",
        },
      ],
    }
  }
/>
`;

exports[`should not render props / methods section if there is no content 1`] = `
<div
  id="Test-container"
>
  <header>
    <_class
      className=""
      level={2}
      slug="test"
    >
      Test
    </_class>
    <div>
      test
    </div>
    <div>
      <_class
        href="#!/Test"
      >
        Open isolated ⇢
      </_class>
    </div>
  </header>
  <div>
    <JsDoc />
  </div>
</div>
`;

exports[`should render both props and methods section 1`] = `
<div
  id="Test-container"
>
  <header>
    <_class
      className=""
      level={2}
      slug="test"
    >
      Test
    </_class>
    <div>
      test
    </div>
    <div>
      <_class
        href="#!/Test"
      >
        Open isolated ⇢
      </_class>
    </div>
  </header>
  <div>
    <JsDoc />
  </div>
  <div>
    <h3>
      Props
    </h3>
    <div>
      prop
    </div>
  </div>
  <div>
    <h3>
      Methods
    </h3>
    <div>
      method
    </div>
  </div>
</div>
`;

exports[`should render component in isolation mode 1`] = `
<div
  id="Test-container"
>
  <header>
    <_class
      className=""
      level={2}
      slug="Test"
    >
      Test
    </_class>
    <div>
      test
    </div>
    <div>
      <_class
        href=""
      >
        ← Back
      </_class>
    </div>
  </header>
  <div>
    <JsDoc />
  </div>
</div>
`;

exports[`should render component not in the isolation mode by default 1`] = `
<div
  id="Test-container"
>
  <header>
    <_class
      className=""
      level={2}
      slug="test"
    >
      Test
    </_class>
    <div>
      test
    </div>
    <div>
      <_class
        href="#!/Test"
      >
        Open isolated ⇢
      </_class>
    </div>
  </header>
  <div>
    <JsDoc />
  </div>
</div>
`;

exports[`should render component renderer for component with examples 1`] = `
<_class
  description="Bar"
  examples={
    <Examples
      examples={
        Array [
          Object {
            "content": "<button>OK</button>",
            "evalInContext": [Function],
            "type": "code",
          },
          Object {
            "content": "Hello *world*!",
            "type": "markdown",
          },
        ]
      }
      name="Foo"
    />
  }
  isolated={false}
  metadata={Object {}}
  methods={false}
  name="Foo"
  pathLine="foo/bar.js"
  slug="foo"
/>
`;

exports[`should render component renderer for component with methods 1`] = `
<_class
  description="Bar"
  examples={
    <Component
      name="Foo"
    />
  }
  isolated={false}
  metadata={Object {}}
  methods={
    <_class
      methods={
        Array [
          Object {
            "description": "Sets the counter to a particular value.",
            "name": "set",
            "params": Array [
              Object {
                "description": "New value for the counter.",
                "name": "newValue",
                "type": Object {
                  "name": "Number",
                },
              },
            ],
            "returns": null,
          },
        ]
      }
    />
  }
  name="Foo"
  pathLine="foo/bar.js"
  slug="foo"
/>
`;

exports[`should render component renderer for component with props 1`] = `
<_class
  description="Bar"
  examples={
    <Component
      name="Foo"
    />
  }
  isolated={false}
  metadata={Object {}}
  methods={false}
  name="Foo"
  pathLine="foo/bar.js"
  props={
    <_class
      props={
        Object {
          "children": Object {
            "description": "Button label.",
            "required": true,
            "type": Object {
              "name": "string",
            },
          },
        }
      }
    />
  }
  slug="foo"
/>
`;

exports[`should render component renderer with example placeholder 1`] = `
<_class
  description="Bar"
  examples={
    <Component
      name="Foo"
    />
  }
  isolated={false}
  metadata={Object {}}
  methods={false}
  name="Foo"
  pathLine="foo/bar.js"
  slug="foo"
/>
`;

exports[`should render methods section 1`] = `
<div
  id="Test-container"
>
  <header>
    <_class
      className=""
      level={2}
      slug="test"
    >
      Test
    </_class>
    <div>
      test
    </div>
    <div>
      <_class
        href="#!/Test"
      >
        Open isolated ⇢
      </_class>
    </div>
  </header>
  <div>
    <JsDoc />
  </div>
  <div>
    <h3>
      Methods
    </h3>
    <div>
      test
    </div>
  </div>
</div>
`;

exports[`should render props section 1`] = `
<div
  id="Test-container"
>
  <header>
    <_class
      className=""
      level={2}
      slug="test"
    >
      Test
    </_class>
    <div>
      test
    </div>
    <div>
      <_class
        href="#!/Test"
      >
        Open isolated ⇢
      </_class>
    </div>
  </header>
  <div>
    <JsDoc />
  </div>
  <div>
    <h3>
      Props
    </h3>
    <div>
      test
    </div>
  </div>
</div>
`;
